indexer-app/src/cmr/indexer/data/concepts/service.clj

```diff
@@ -23,6 +23,15 @@
         supported-formats (get-in service [:ServiceOptions :SupportedFormats])]
     (> (count supported-formats) 1)))
 
+(defn- has-spatial-subsetting?
+  "Returns true if the given service has a defined SubsetType with one of its
+  values being 'spatial'."
+  [context service-concept]
+  (let [service (concept-parser/parse-concept context service-concept)
+        {{subset-type :SubsetType} :ServiceOptions} service]
+    (and (seq subset-type)
+         (contains? (set subset-type) "Spatial"))))
```

Comment:
This is still one of the most confusing parts of Clojure for me - should I be using `contains?` or `some` for finding a value in a collection? `contains?` needs to build a set, but has constant or logarithmic lookup time, while `some` doesn't have to build a set but needs to perform a linear search.

I assume that `contains?` should almost always be preferred then, so why does the documentation for `some` specifically list "One common idiom... `(some #{:fred} coll)`"?

Reply:
Note that `(contains? (set subset-type) "Spatial"))` returns a boolean; `(some #{"Spatial"} subset-type)` returns a matching value. One way to make the latter return a bool is `(some? (some #{"spatial"} subset-type))` ... awkward. Another way to do this would be `(boolean (some #(= % "spatial") subset-type))` (the `boolean` is needed to convert the non-matching `nil` to `false`).

For my own view of readability, `(contains? (set subset-type) "Spatial"))` beats the others ...

For performance, let's see ... 

On a per-function call basis:

`contains?`: ~0.06 msecs
`some?`/`some`: ~ 0.05 msecs
`boolean`/`some`: ~0.15-0.4 msecs

For the last one I gave a range because its performance fluctuates pretty widely.

10 millions calls to the following with small data sets gives:

`contains?`: ~4.5 seconds
`some?`/`some`: ~3.2 seconds
`boolean`/`some`: ~3.5 seconds

Given the similarity of performance (especially on the per-single-call basis), I'd probably lean towards `contains?` ... but it's up to you guys.

Reply:
We only have three possible values in the set, I don't think we care about performance in this case. `contains?` returns true, so a good choice.